File: day9/day9.go
package main

import (
	"io/ioutil"
	"fmt"
)

func main() {
	data, err := ioutil.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	input := string(data)

	decompressed := 0
	for i := 0; i < len(input); i++ {
		if input[i] == '(' {
			group := ""
			i++
			for input[i] != ')' {
				group += string(input[i])
				i++
			}
			length, repeat := 0, 0
			if _, err := fmt.Sscanf(group, "%dx%d", &length, &repeat); err != nil {
				panic(fmt.Errorf("%v while scannning %s", err, group))
			}

			i += length
			decompressed += length * repeat
		} else {
			decompressed++
		}
	}
	fmt.Println("Part 1:", decompressed)

	for i := len(input)-1; i >= 0; i-- {

	}
}

File: day9/day9.cs
using System;

class Day9
{
    static void Main()
    {
        string input = System.IO.File.ReadAllText("input.txt");

        int decompressed = 0;
        for (int i = 0; i < input.Length; i++)
        {
            if (input[i] == '(')
            {
                var buf = new System.Text.StringBuilder();
                i++;
                for (;input[i] != ')';)
                {
                    buf.Append(input[i]);
                    i++;
                }

                var group = buf.ToString().Split('x');
                int length = Convert.ToInt32(group[0]);
                int repeat = Convert.ToInt32(group[1]);

                i += length;
                decompressed += length * repeat;
            }
            else
            {
                decompressed++;
            }
        }

        Console.WriteLine("Part 1: {0}", decompressed);
    }
}